### PR TITLE
[ENH] new navigation and change internal organisation

### PR DESCRIPTION
**Pull request by Remi-Gau**
- to match proposal from @robertoostenveld from https://github.com/robertoostenveld/bids-website/tree/main-v3.1
- empty index.md have been added to most subfolders (see #461)
- submodules have been moved either in the root folder when not used directly in the doc or into the most appropriate subfolders and renamed `src` where possible.
- bids apps submodule moved into `data` as its `config.yml` file is required to build the bids app page.
- examples submodule moved into `data` as its dataset listing file is required to build the datasets page.
- faqs were moved from one file per question to a single file for all the questions of a section
- content from the bids starter kit and the extension-guide submodule was transferred

## TODO

- [x] fix some page titles so they are properly capitalized in the navbar

### CODE REVIEW

**Comment by robertoostenveld** [comment]
I am happy to see how this works out. The `medics serve` command also shows a bunch of useful warning messages:

```
WARNING -  A reference to 'standards/bids_stats_model/src/specification/index.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/motivation.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/walkthrough-intro.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/walkthrough-1.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/integration.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/legacy.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/spec.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/reference.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/model-zoo.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'standards/bids_stats_model/src/specification/default_model.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/schema.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/how-the-schema-works.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/schema_meta.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/schema_objects.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/schema_rules.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'specification/schema/README.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'getting_started/src/src/index.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'getting_started/src/src/links.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'getting_started/src/src/dependencies.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'getting_started/src/src/publications.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'getting_started/src/src/talks.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/general.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/eeg.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/mri.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/phenotype.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/bids_extensions.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'faq/bids_apps.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'specification/code_of_conduct.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/index.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/general_guidelines.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/guide.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  A reference to 'extensions/src/docs/submission.md' is included in the 'nav' configuration, which is not found in the documentation files.
WARNING -  Doc file 'index.md' contains a link './getting_started/src/src/index.md', but the target 'getting_started/src/src/index.md' is not found among documentation files.
WARNING -  Doc file 'standards/bids_app_specification/scope.md' contains a link './specification/release_notes.md#[0.1.0.dev]', but the target 'standards/bids_app_specification/specification/release_notes.md' is not found among documentation files.
```

which point to the menu items that are not yet fulfilled. Since no md files have been deleted (and hence no documentation has been lost) in this pR, I suggest that you merge it with main and that we take the next step in completing the linkage between navigation and md files (by "git moving" them into the right place).

Is there a specific section that I can help on?

**Comment by Remi-Gau** [comment]
Am not seeing those warnings on read the docs: https://app.readthedocs.org/projects/bids-website/builds/25153345/

before running mkdocs serve:

- update the submodules (should fix all issues related to missing pages in some `src` folder)

```bash
git submodule update --init --recursive && git submodule update --recursive
```

- make sure you installed all the requirements

```bash
python -m pip install --exists-action=w --no-cache-dir -r requirements.txt
```

- generate all content (should fix things like missing FAQ pages)

```bash
make update
```

**Comment by robertoostenveld** [comment]
The new version of the website including #463, #464 and #465 is rendered on https://static.robertoostenveld.nl/bids-website-462+463+464+465. It starts looking quite complete, but many small edits are still needed and lead paragraphs have to be made consistent to ensure that the left menu appears and works consistently. 

Broken links and images need to be checked and fixed, that is also where the mkdocs error messages are helpful.

**Comment by Remi-Gau** [comment]
some of the links need to be fixed in the bids example repo

File: docs/extensions/general-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+Try not to deviate from BIDS conventions in your extension.
+
+### Be consistent with other BEPs
+
+A common dictionary (BIDS keys) is what makes BIDS successful,
+it is thus essential to not create many new entities.
+Many of the current BEPs have developed useful terms that we recommend here:
+
+| Entity                                               | BEP(s)                                                                                   | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      |
```
Comment (Remi-Gau) [inline docs/extensions/general-guidelines.md:140]: divide this table into entities that are already merged and those that are not

possibly generate the first from from the schema and point to the relevant pages in the bids spec

File: docs/extensions/general-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+No. This information is complementary.
+
+> How do I know if the metadata I want to store should be put in the BIDS-derivatives BEP or in a JSON-LD file?
+
+It will never go in JSON-LD. BIDS-derivatives and BIDS-Prov are superpositions, they are non-overlapping. You don't have to include BIDS-Prov.
+
+> If I don’t use BIDS-Prov how can I describe the data workflow?
+
+A lightweight version is to use a description.tsv file (see common [derivatives](https://bids-specification.readthedocs.io/en/stable/05-derivatives/02-common-data-types.html))
```
Comment (Remi-Gau) [inline docs/extensions/general-guidelines.md:189]: move this section into the FAQ

File: docs/extensions/general-guidelines.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+preferably also with a mix of more junior and more senior contributors.
+You may also consider requesting explicit support letters from external labs.
+
+### How to turn on email notifications about suggestions and comments for Google Docs
+
+![notifications_1](../assets/img/notifications_1.png)
+
+![notifications_2](../assets/img/notifications_2.png)
+
+![notifications_3](../assets/img/notifications_3.png)
```
Comment (Remi-Gau) [inline docs/extensions/general-guidelines.md:271]: Create a how-to section and put this into it.

**Comment by Remi-Gau** [comment]
@robertoostenveld @iamdamion @bendhouseart @ericearl 

I don't think that anyone wants to review this monster.

So I will merge, close any issues that this resolves and open a series of new ones.

File: docs/extensions/general-guidelines.md (hunk 2 above)
Comment (Remi-Gau) [reply docs/extensions/general-guidelines.md:189]: done

File: docs/extensions/general-guidelines.md (hunk 3 above)
Comment (Remi-Gau) [reply docs/extensions/general-guidelines.md:271]: moved to FAQ

**Pull request merged**